src/environment.rs

```diff
@@ -20,10 +37,13 @@ impl Environment {
         let value_map = HashTrieMap::default();
         for i in 1..=num_args {
             let par_i = Path::LocalVariable { ordinal: i };
-            // todo: figure out how to get a source span for each parameter definition
             value_map.insert(par_i, abstract_value::TOP);
         }
-        Environment { value_map }
+        Environment {
```

Comment:
Environment { value_map, ..Default::default() }

Is nice in some ways, although then it doesn't force you to think about how new fields should be defined.

Reply:
Since entry_conditions does not take the default value, it is slightly nicer to explicitly list all of the fields, but it is nice to know that there is such a syntax. I'll try to remember it.